### PR TITLE
Add Group Management Events - Add NewUAC Description for User Management Events

### PR DESCRIPTION
**Pull request by janniten**
Added Group Management Events and Events 4798 and 4799 (User and Group enumeration)

Event | Description
--|--|
4727|A security-enabled global group was created.
4728|A member was added to a security-enabled global group.
4729|A member was removed from a security-enabled global group.
4730|A security-enabled global group was deleted.
4731|A security-enabled local group was created
4732|A member was added to a security-enabled local group.
4733|A member was removed from a security-enabled local group.
4734|A security-enabled local group was deleted.
4735|A security-enabled local group was changed.
4737|A security-enabled global group was changed.
4754|A security-enabled universal group was created.
4755|A security-enabled universal group was changed.
4756|A member was added to a security-enabled universal group.
4757|A member was removed from a security-enabled universal group.
4758|A security-enabled universal group was deleted.
4764|A group\'s type was changed.
4798|A user's local group membership was enumerated.
4799|A security-enabled local group membership was enumerated.

In order to map correclty this event a new field group.domain was added to the ECS group schema
(https://github.com/elastic/ecs/pull/547)

Added addUACDescription function in order to translate to a human readable form the  flags in winlog.event_data.NewUacValue field. A new field winlog.event_data.NewUACList is created and contains a list of decoded flags from the hex value in winlog.event_data.NewUACList
For example
winlog.event_data.NewUacValue -> 0x15 is translated to winlog.event_data.NewUACList -> SCRIPT,LOCKOUT 
Also converts the winlog.event_data.UserAccountControl to a list of values

Cosmetic change: fix the order of events 4767 and 4781 and some space fixing

Ingesting this events we can have information about group who performed group changes and what the changes are.

Group_Managment - Kibana

### CODE REVIEW

**Comment by janniten** [comment]
@andrewkroh 
Group Management Events and some improvements to User Management Events added in this PR

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 1]
```diff
@@ -27,15 +55,34 @@ var security = (function () {
         "4720": "added-user-account",
         "4722": "enabled-user-account",
         "4723": "changed-password",
-        "4724": "reset-password",
+        "4724": "reseted-password",
```
[the remark this answers is not in the document]
Comment (janniten) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: Fixed :)

**Comment by andrewkroh** [comment]
This dashboard looks great. Would you like to add that to the PR?

`.\winlogbeat.exe export dashboard -id [uuid-of-the-dashboard-taken-from-the-url] > my-dashboard.json`

**Comment by janniten** [comment]
> This dashboard looks great. Would you like to add that to the PR?
> 
> `.\winlogbeat.exe export dashboard -id [uuid-of-the-dashboard-taken-from-the-url] > my-dashboard.json`
> 
> (I haven't tried this on Windows, but I think it should work. This assumes Kibana is a `localhost:5601`.)
Yes! I'll add the dashboard to the PR today :)

**Comment by janniten** [comment]
@andrewkroh I have already exported the dashboard in a .json.
Shall I put the exported dashboards in the folder beats/winlogbeat/_meta/kibana/7/dashboard ?
I have included also a similar dashboard for user managment events.

I'll wait for your response and then I'll update the PR.

**Comment by andrewkroh** [comment]
>  Shall I put the exported dashboards in the folder beats/winlogbeat/_meta/kibana/7/dashboard ?

Since these dashboards are dependent on the transformations done in the module I think they belong in the module dir at `x-pack/winlogbeat/module/security/_meta/kibana/7/dashboard`. Thanks

**Comment by janniten** [comment]
Done! 
I put in the PR dashboards for User Management Events (https://github.com/elastic/beats/pull/13530) and for Group Management Events. 
In both cases I have included two versions: one using Visual Builder metrics that are nicer & colorful  (but may require to adjust the interval size in the Panel Options according to your own needs) and one with normal metric visualizations that can be used without adjusting any parameter.

**Comment by janniten** [comment]
@andrewkroh, It is not clear for me which is the error and If I have to correct something
It seems some problem with the Json related to the new dashboards.
Shall I remove the dashboards in order to unblock the PR?
Once this PR is closed I'm preparing a new one with the events of computer management (4741,4742 and 4743) and the events related to audit change/deletion/errors (1100,1102,1104,1105,1108 and 4719)

Thank you, 
Regards
Ana

**Comment by janniten** [comment]
Hi @leehinman 
I have fixed the spacing problems that were found.
Please feel free to add the golden files and ask me any further modification.
Thank you!

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 2]
```diff
@@ -1054,7 +1101,7 @@ var security = (function () {
         }
         var eventActionDescription = eventActionTypes[code];
         if (eventActionDescription) {
-            evt.Put("event.action", eventActionDescription);
+          evt.Put("event.action", eventActionDescription);
```
[the remark this answers is not in the document]
Comment (janniten) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: fixed

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 3]
```diff
@@ -1234,78 +1317,143 @@ var security = (function () {
...
+        .Add(copySubjectUserLogonId)
+        .Add(copyTargetUserToGroup)
+        .Add(renameCommonAuthFields)
+        .Add(addActionDesc)
         .Build();
 
     return {
-        // 4624 - An account was successfully logged on.
-        4624: logonSuccess.Run,
+
```
[the remark this answers is not in the document]
Comment (janniten) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: fixed

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 4]
```diff
@@ -1122,6 +1203,7 @@ var security = (function () {
         })
         .Build();
 
+
```
[the remark this answers is not in the document]
Comment (janniten) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: fixed

**Comment by leehinman** [comment]
jenkins, test this

**Comment by leehinman** [comment]
jenkins, test this

**Comment by leehinman** [comment]
@janniten I got the golden files added and updated the dashboards to pass the tests.  Can you double check the dashboards to make sure I didn't break them?

**Comment by janniten** [comment]
@leehinman, I've tried to import the dashboards and I get this error
To test it I downloaded the raw json and I tried to import the dashboard using the winlogbeat setup --dashboard. Is the way I've tested correct?

error loading C:\Program Files\Winlogbeat\kibana\7\dashboard\db_test.json: failed to parse field [dashboard.kibanaSavedObjectMeta.searchSourceJSON] of type [text] in document with id 'siem:dashboard:bb858830-f412-11e9-8405-516218e3d268'. Preview of field's value: '{filter=[], query={query=, language=kuery}}'. Response: {"objects":[{"id":"bb858830-f412-11e9-8405-516218e3d268","type":"dashboard","error":{"message":"failed to parse field [dashboard.kibanaSavedObjectMeta.searchSourceJSON] of type [text] in document with id 'siem:dashboard:bb858830-f412-11e9-8405-516218... (truncated)

Thank you!

**Comment by janniten** [comment]
> @janniten I got the golden files added and updated the dashboards to pass the tests. Can you double check the dashboards to make sure I didn't break them?

@leehinman, Maybe I can remove the dashboards so the PR can be merge and create a separate PR for the dashboards. What do you think?

**Comment by leehinman** [comment]
> > @janniten I got the golden files added and updated the dashboards to pass the tests. Can you double check the dashboards to make sure I didn't break them?
> 
> @leehinman, Maybe I can remove the dashboards so the PR can be merge and create a separate PR for the dashboards. What do you think?

@janniten Let me see if I can test the dashboard, @andrewkroh gave me some pointers I'd like to try out.

**Comment by janniten** [comment]
> @janniten Let me see if I can test the dashboard, @andrewkroh gave me some pointers I'd like to try out.

Hi @leehinman, Can I help/do something more?

**Comment by leehinman** [comment]
@janniten I haven't (yet) figured out how to get the dashboards working.  If you are still willing maybe we should split the PR.  I'll keep working on the dashboards.

**Comment by janniten** [comment]
> @janniten I haven't (yet) figured out how to get the dashboards working. If you are still willing maybe we should split the PR. I'll keep working on the dashboards.

@leehinman I removed the dashboards. I think now is ok.
Once merged I'll create two more PRs: one with the dashboards and one for others windows events I've been working

**Comment by leehinman** [comment]
jenkins, test this

**Comment by janniten** [comment]
> jenkins, test this

@leehinman , shall I modify something?  CI Build failed

**Comment by janniten** [comment]
@leehinman  one question. Events added in this PR are not listed in the release notes
https://www.elastic.co/guide/en/beats/libbeat/current/release-notes-7.6.0.html

Do you know why?
Thank you

**Comment by leehinman** [comment]
> @leehinman one question. Events added in this PR are not listed in the release notes
> https://www.elastic.co/guide/en/beats/libbeat/current/release-notes-7.6.0.html
> 
> Do you know why?
> Thank you

@janniten I'm sorry, this is my fault.  I didn't catch that there wasn't a Changelog entry, and the info in the release notes is populated from the Changelog.  I did check and the code is in the release.
```
    [Winlogbeat] Add Group Management Events - Add NewUAC Description for User Management Events (#14299) (#15153)

    * Added Group Management Events
    * Added User and Group Enumeration
    * Added New UAC Description

    (cherry picked from commit 8e31628955f89ab550e33a1ae1cc750b850a29ab)

    Co-authored-by: Anabella Cristaldi <33020901+janniten@users.noreply.github.com>
```

**Pull request merged**